src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -1065,6 +1098,147 @@ protected Map<String, Object> getFirstInnerHit(final Map<String, Object> searchR
         return (Map<String, Object>) hits2List.get(0);
     }
 
+    /**
+     * Get InnerHits from the search hits
+     *
+     * @param searchResponseAsMap Complete search response as a map
+     * @return list of innerHits where each element in the hit corresponds to a SearchHit
+     */
+    @SuppressWarnings("unchecked")
+    protected List<Object> getInnerHitsFromSearchHits(final Map<String, Object> searchResponseAsMap) {
+        Map<String, Object> hits1map = (Map<String, Object>) searchResponseAsMap.get("hits");
+        List<Map<String, Object>> hits2List = (List<Map<String, Object>>) hits1map.get("hits");
+        assertTrue(hits2List.size() > 0);
+        List<Object> innerHits = new ArrayList<>();
+        for (Map<String, Object> hitObject : hits2List) {
+            innerHits.add(hitObject.get("inner_hits"));
+        }
+        return innerHits;
+    }
+
+    /**
+     * Get Total and actual count of each inner hit field
+     *
+     * @param innerHits List of inner hits
+     * @param  fieldNames list of field Names of which total count need to be retrived
+     * @return Map of fieldName to totalCount of inner_hit from each individual hit
+     */
+    protected Map<String, Map<String, ArrayList<Integer>>> getInnerHitsCountsOfNestedField(
+        List<Object> innerHits,
+        List<String> fieldNames
+    ) {
+        Map<String, Map<String, ArrayList<Integer>>> countsPerFieldMap = new HashMap<>();
+
+        for (Object innerHit : innerHits) {
+            Map<String, Object> hits = (Map<String, Object>) innerHit;
+            for (String fieldName : fieldNames) {
+                Map<String, Object> searchHits = (Map<String, Object>) hits.get(fieldName);
+                Map<String, Object> fieldHits = (Map<String, Object>) searchHits.get("hits");
+
+                Map<String, Object> total = (Map<String, Object>) fieldHits.get("total");
+                int totalCount = (Integer) total.get("value");
+
+                List<Object> innerHitsOfField = (List<Object>) fieldHits.get("hits");
+                int actualCount = innerHitsOfField.size();
+
+                countsPerFieldMap.computeIfAbsent(fieldName, k -> new HashMap<>())
+                    .computeIfAbsent("total", k -> new ArrayList<>())
+                    .add(totalCount);
+                countsPerFieldMap.computeIfAbsent(fieldName, k -> new HashMap<>())
+                    .computeIfAbsent("actual", k -> new ArrayList<>())
+                    .add(actualCount);
+            }
+        }
+        return countsPerFieldMap;
+    }
+
+    /**
+     * Create Sort Builders to be applied with QueryBuilder
+     *
+     * @param fieldSortOrderMap Map of fieldName of the sort and its order
+     * @param  isSortByScore boolean flag to notify if sort condition is on _score
+     * @return List of SortBuilders
+     */
+    protected List<SortBuilder<?>> createSortBuilders(Map<String, SortOrder> fieldSortOrderMap, boolean isSortByScore) {
```

Comment:
This method was previously present in HybridQuerySortIT. I have moved it in BaseNeuralSearchIT for reusing this method in other test classes.